### DISCUSSION TITLE
jakarta migration to support tomcat 10

### DISCUSSION THREAD

**Discussion by dreddyio** [mitre/HTTP-Proxy-Servlet#223] [Q&A]
hi, I see that the current implementation of the servlet spec is using javax.* packages. Is there a timeline to migrate to jakarta.* spec as tomcat 10 does not support javax.* spec anymore?

**Comment by dsmiley** [comment]
I'm not too familiar with the latest goings on in this space so it would be helpful if you could point me at an article/post or two.
Presumably this would be an either-or and thus after this change, the servlet wouldn't work in older servlet engine versions?

**Reply by dreddyio** [reply]
https://tomcat.apache.org/download-10.cgi

Users of Tomcat 10 onwards should be aware that, as a result of the move from Java EE to Jakarta EE as part of the transfer of Java EE to the Eclipse Foundation, the primary package for all implemented APIs has changed from javax.* to jakarta.*. This will almost certainly require code changes to enable applications to migrate from Tomcat 9 and earlier to Tomcat 10 and later.

**Comment by EdgewareRoad** [comment]
And, for those of use who integrate HTTP-Proxy-Servlet into Spring Boot, the option to migrate and use in Production just isn’t there yet as Spring Boot 6 won’t be out until (projected) Q4 2022.
A Java 17 and Jakarta EE 9 baseline for Spring Framework 6<https://spring.io/blog/2021/09/02/a-java-17-and-jakarta-ee-9-baseline-for-spring-framework-6>

Appreciate that there may well be a need for this but looks like at least an irritating year or so where, if both javax.servlet and jakarta.servlet are to be supported, we need two projects, one a fork of the other but with differences at the namespace level and for any changes to be merged into both.

Right now, I’d advise against a move now but would recommend that whoever needs this to create their own fork of the project to refactor and test this under Tomcat 10. At a later stage, this could be considered as a merge back into the main project perhaps when both Tomcat 10 (and other servlet engines supporting Jakarta EE 9) and Spring Boot 6 have been out a while and proven their maturity for new projects and technology refresh/updates for existing code.

Unless otherwise stated, this email has been sent from Fujitsu Services Limited (registered in England No 96056); Fujitsu EMEA PLC (registered in England No 2216100) both with registered offices at: 22 Baker Street, London W1U 3BW; PFU (EMEA) Limited, (registered in England No 1578652) registered offices at: Belmont, Belmont Road, Uxbridge, England, UB8 1HE and Fujitsu Research of Europe Ltd (registered in England No. 4153469) 4th Floor, Building 3, Hyde Park Hayes, 11 Millington Road, Hayes, UB3 4AZ.

This email is only for the use of its intended recipient. Its contents are subject to a duty of confidence and may be privileged. Fujitsu does not guarantee that this email has not been intercepted and amended or that it is virus-free.
